test/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration.Tests/ExtensionsTests/StringExtensionTest.cs

```diff
@@ -45,60 +45,94 @@ public static IEnumerable<object[]> GetTestCasesForToCamelCaseShouldSucceed()
             };
         }
 
-        public static IEnumerable<object[]> GetTestCasesForRemoveRoslynDuplicateStringShouldSucceed()
+        public static IEnumerable<object[]> GetTestCasesForRemoveRoslynDuplicateStringShouldUpdateCorrectly()
         {
+            yield return new object[]
+            {
+                "String with even length starting from -",
+                "abc-ef-efg",
+                "abc-ef-efg"
+            };
+
+            yield return new object[]
+            {
+                "String with odd length starting from -",
+                "abc-de-de",
+                "abc-de"
+            };
+
+            yield return new object[]
+            {
+                "String staring with @ and odd length starting from -",
+                "@abc-de-de",
+                "@abc-de"
+            };
+
+            yield return new object[]
+            {
+                "String staring with @ and odd length starting from - and no duplicates",
```

Comment:
Can we say "string prefixed with @ with the special-character-containing part having even length starting with - without duplicates"?